core/templates/dev/head/pages/exploration-player-page/services/learner-answer-info.service.ts

```diff
@@ -0,0 +1,126 @@
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview Service for learner answer info.
+ */
+
+require(
+  'pages/exploration-player-page/services/answer-classification.service.ts');
+require('domain/statistics/LearnerAnswerDetailsBackendApiService.ts');
+
+angular.module('oppia').factory('LearnerAnswerInfoService', [
+  'AnswerClassificationService', 'LearnerAnswerDetailsBackendApiService',
+  'INTERACTION_IDS_WITHOUT_ANSWER_DETAILS',
+  function(
+      AnswerClassificationService, LearnerAnswerDetailsBackendApiService,
+      INTERACTION_IDS_WITHOUT_ANSWER_DETAILS) {
+    var submittedAnswerInfoCount = 0;
+    var currentEntityId = null;
+    var stateName = null;
+    var interactionId = null;
+    var currentAnswer = null;
+    var currentInteractionRulesService = null;
+    var canAskLearnerForAnswerInfo = false;
+    var visitedStates = [];
+    var probabilityIndexes = {
+      // The probability that a request for explanation of the answer that is
+      // submitted by the learner. There are three different probabilities
+      // based on the outcome of the answer.
+      // The probability index when the outcome is equal to the default outcome
+      // for an interaction.
+      type_a: 0.25,
+      // The probability index when the outcome is marked as correct i.e
+      // labelled_as_correct property is true.
+      type_b: 0.10,
+      // The probability index when the outcome is not the default outcome
+      // and it is not marked as correct i.e. it is any general outcome.
+      type_c: 0.05
+    };
+
+    var getRandomProbabilityIndex = function() {
+      var min = 0;
+      var max = 100;
+      return (Math.floor(Math.random() * (max - min + 1) ) + min) / 100;
+    };
+
+    return {
+      initLearnerAnswerInfoService: function(
+          entityId, state, answer, interactionRulesService) {
+        currentEntityId = entityId;
+        currentAnswer = answer;
+        currentInteractionRulesService = interactionRulesService;
+        stateName = state.name;
+        interactionId = state.interaction.id;
+        var defaultOutcome = state.interaction.defaultOutcome;
+
+        if (submittedAnswerInfoCount === 2) {
+          return;
+        }
+
+        if (!state.solicitAnswerDetails) {
+          return;
+        }
+
+        if (INTERACTION_IDS_WITHOUT_ANSWER_DETAILS.indexOf(
+          interactionId) !== -1) {
+          return;
+        }
+
+        if (visitedStates.indexOf(stateName) !== -1) {
+          return;
+        }
+
+        var classificationResult = (
+          AnswerClassificationService.getMatchingClassificationResult(
+            stateName, state.interaction, answer,
+            interactionRulesService));
+        var outcome = classificationResult.outcome;
+        var thresholdProbabilityIndex = null;
+        var randomProbabilityIndex = getRandomProbabilityIndex();
+        /* eslint-disable dot-notation */
```

Comment:
Why do you need this? We use dot notation elsewhere...

Reply:
Removed